File: ios/ReactNativeEventEmitter.m
//
//  ReactNativeEventEmitter.m
//  fraa
//
//  Created by Donbosco on 8/28/20.
//


#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>
#import <React/RCTEventEmitter.h>

@interface

RCT_EXTERN_MODULE(ReactNativeEventEmitter, RCTEventEmitter)

RCT_EXTERN_METHOD(supportedEvents)

// - (dispatch_queue_t)methodQueue
// {
//   return dispatch_queue_create("com.facebook.React.AsyncLocalStorageQueue", DISPATCH_QUEUE_SERIAL);
// }
 + (BOOL)requiresMainQueueSetup
 {
 //  return YES;  // only do this if your module initialization relies on calling UIKit!
    return NO;
 }
@end

File: ios/ReactNativeEventEmitter.swift
//
//  ReactNativeEventEmitter.swift
//  fraa
//
//  Created by Donbosco on 8/28/20.
//


import UIKit
import Foundation
@objc(ReactNativeEventEmitter)
open class ReactNativeEventEmitter: RCTEventEmitter {
  override init() {
      super.init()
      EventEmitter.sharedInstance.registerEventEmitter(eventEmitter: self)
  }
  @objc
 static var serialQueue = DispatchQueue(label: "ReactNativeEventEmitter.serial.queue")
  
  /// Base overide for RCTEventEmitter.
  ///
  /// - Returns: all supported events
  @objc open override func supportedEvents() -> [String] {
      return EventEmitter.sharedInstance.allEvents
  }
}

